kinesis/kinesis_test.go

```diff
@@ -0,0 +1,129 @@
+package kinesis
+
+import (
+    "os"
+    "time"
+    "math/rand"
+    "testing"
+
+    "github.com/aws/amazon-kinesis-firehose-for-fluent-bit/plugins"
+    "github.com/aws/aws-sdk-go/service/kinesis"
+    "github.com/stretchr/testify/assert"
+    "github.com/sirupsen/logrus"
+    "github.com/golang/mock/gomock"
+    "github.com/aws/aws-sdk-go/aws"
+    "github.com/aws/amazon-kinesis-streams-for-fluent-bit/kinesis/mock_kinesis"
+    fluentbit "github.com/fluent/fluent-bit-go/output"
+)
+
+// newMockOutputPlugin creates an mock OutputPlugin object
+func newMockOutputPlugin(client *mock_kinesis.MockPutRecordsClient) (*OutputPlugin, error) {
+    records := make([]*kinesis.PutRecordsRequestEntry, 0, 500)
+
+    timer, _ := plugins.NewTimeout(func (d time.Duration) {
+        logrus.Errorf("[kinesis] timeout threshold reached: Failed to send logs for %v", d)
+        logrus.Errorf("[kinesis] Quitting Fluent Bit")
+        os.Exit(1)
+    })
+
+    seededRand := rand.New(rand.NewSource(time.Now().UnixNano()))
+    b := make([]byte, 8)
+    random := &random{
+        seededRandom:   seededRand,
+        buffer:         b,
+    }
+
+    return &OutputPlugin{
+        region:                         "us-west-2",
+        stream:                         "stream",
+        client:                         client,
+        records:                        records,
+        dataKeys:                       "",
+        partitionKey:                   "",
+        lastInvalidPartitionKeyIndex:   -1,
+        backoff:                        plugins.NewBackoff(),
+        timer:                          timer,
+        PluginID:                       0,
+        random:                         random,
+    }, nil
+}
+
+func TestStringOrByteArray(t *testing.T) {
```

Comment:
Please combine `TestStringOrByteArray` and `TestStringOrByteArrayWithNil` into 1 test using table driven tests: https://github.com/golang/go/wiki/TableDrivenTests

Reply:
Updated!